### ISSUE TITLE
EnvironmentModifications.from_sourcing_file() hangs.

### ISSUE THREAD

**Issue by burgreen** [spack/spack#8930]
`spack install openfoam-com` on linux-centos6-x86_64 system hangs during installation. 

If the post-install `try` block related to the `from_sourcing_file` call is commented out, installation proceeds fine to completion.

**Comment by adamjstewart** [comment]
@olesenm

**Comment by olesenm** [comment]
Possibly @alalazo - Greg already discussed this with me per email. The EnvironmentModifications seems to be doing something odd on his system (I suspect). At the very start, I avoided EnvironmentModifications since it would bomb out on me. In the meantime it worked OK, but it is still a point of suspicion for me. Of course, by all means not ruling out that there is something wrong on the OpenFOAM side.

**Comment by burgreen** [comment]
Turns out `from_sourcing_file()` was hanging at the `proc.wait()` command. Is its inclusion necessary or in error?

**Comment by burgreen** [comment]
Python docs say this:

Popen.wait()
    Wait for child process to terminate. Set and return returncode attribute.
    Warning
    This will deadlock when using stdout=PIPE and/or stderr=PIPE and the child process generates enough output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data. Use communicate() to avoid that.

**Comment by burgreen** [comment]
Commenting out the `proc.wait()` leads to this:
```==> Successfully installed openfoam-com
  Fetch: 0.07s.  Build: 57m 7.99s.  Total: 57m 8.06s.
[+] /var/tmp/burgreen/dev/stack/opt/spack/linux-centos6-x86_64/gcc-7.3.0-spack/openfoam-com-1806-d4ham6y5733s337ke2l7veo54444y3po
Traceback (most recent call last):
  File "<string>", line 1, in <module>
IOError: [Errno 32] Broken pipe
```
I'm not sure if it was properly installed or not.

**Comment by olesenm** [comment]
This sounds like you are probably onto the right solution. If you modify your EnvironmentModifications to use communicate instead, I'd hope that this solves the problem. Do you have any infinite loops hidden away in your bashrc + OpenFOAM bashrc that could explain part of this?

**Comment by burgreen** [comment]
`subprocess.Popen.communicate()` seemed to work for me. It fully installed quietly.

**Comment by olesenm** [comment]
@burgreen and @alalazo  - any updates or ideas on how to progress on this?

**Comment by burgreen** [comment]
Replacing `proc.wait()` with `proc.communicate()` in `EnvironmentModifications.from_sourcing_file()` seemed to work for me. Whether this is the proper solution or not is another question.

**Comment by alalazo** [comment]
Closing as solved in #10753. Feel free to reopen if I'm wrong.
